File: sdk/examples/outbound_sales/src/prompt_template_format.py
from .utilities import _read_prompt_from_system_and_id, _format_template_string
from .prompt_formatted import Prompt_Formatted

# FreewayAI is property of DiligentlyAI, Inc. and is used under an MIT license.
# by Ben Parfitt, Feb 2024

def _format_prompt(prompt, **kwargs):

    messages = []
    if "messages" in prompt:
        messages = _format_messages(prompt, prompt["messages"], **kwargs)

    if "prompt_template" in prompt:
        msg = {
            "content": _format_template_string(prompt, prompt["prompt_template"], **kwargs),
            "role": "user"
        }
        if len(messages):
            messages.append(msg)
        else:
            messages = [msg]
        
    return Prompt_Formatted(messages)

def _format_messages(prompt, messages, **kwargs):

    for message in messages:
        message["content"] = _format_template_string(prompt, message["content"], **kwargs)

    return messages

def read_and_format_prompt(system_id, systems_location, prompt_id, **kwargs):
    prompt = _read_prompt_from_system_and_id(system_id, systems_location, prompt_id)
    return _format_prompt(prompt, **kwargs)

def main():
    print(read_and_format_prompt('example', 'better_template_id', type="none here", description="none that I know of"), end="\n\n")
    print(read_and_format_prompt('example', 'better_template_id2', type="none here", description="none that I know of"), end="\n\n")
    print(read_and_format_prompt('example', 'better_template_id3', type="none here", description="none that I know of"), end="\n\n")
    print(read_and_format_prompt('example', 'better_template_id4', type="none here", description="none that I know of"), end="\n\n")

if __name__ == "__main__":
    main()

File: sdk/js/freewayai/src/prompt_template_format.ts
import { _read_prompt_from_system_and_id, _format_template_string} from './utilities';
import { Prompt_Formatted } from './prompt_formatted';

function _format_prompt(prompt: any, variables: Record<string, any>): Prompt_Formatted {

  let messages: any[] = [];
  if (prompt.messages) {
    messages = _format_messages(prompt, prompt.messages, variables);
  }

  if (prompt.prompt_template) {
    const msg = {
      content: _format_template_string(prompt, prompt.prompt_template, variables),
      role: "user",
    };

    if (messages.length) {
      messages.push(msg);
    } else {
      messages = [msg];
    }
  }

  return new Prompt_Formatted(messages);
}

function _format_messages(prompt: any, messages: any[], variables: Record<string, any>): any[] {
  return messages.map(message => ({
    ...message,
    content: _format_template_string(prompt, message.content, variables),
  }));
}

function read_and_format_prompt(systemId: string, systemsLocation: string, promptId: string, variables: Record<string, any>): Prompt_Formatted {
  const prompt = _read_prompt_from_system_and_id(systemId, systemsLocation, promptId);

  return _format_prompt(prompt, variables);
}

export { read_and_format_prompt }

